partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rex/CastTable.kt

```diff
@@ -236,12 +236,12 @@ internal object CastTable {
             )
         }
         register(SMALLINT, DECIMAL_ARBITRARY) { x, _ ->
```

Comment:
With the renaming of `INT_ARBITRARY`, I would've thought you would remove `DECIMAL_ARBITRARY`. Is this reserved for another PR?

Reply:
Another PR. I started to do some of the things listed in the follow-ups, but then the PR got too complicated.